compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1925,6 +1925,26 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     *impl_,
                 );
                 if let Some(assoc_ty) = assoc_ty {
+                    let (assoc_ident, def_scope) =
+                        tcx.adjust_ident_and_get_scope(assoc_ident, adt_def.did(), hir_ref_id);
```

Comment:
This can be used to make the lookup faster than `find_by_name_and_kind`, like what trait associated items do below.

Reply:
The second parameters is wrong. It should be `impl_.def_id`: we rebase hygiene information over the scope that defines `assoc_ty`, which may be unrelated to the adt's scope.